src/figuresClasses.ts

```diff
@@ -1,19 +1,101 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('All sides of triangle must be greater than 0!!!');
+    }
+
+    const max = Math.max(this.a, this.b, this.c);
```

Comment:
DRY - don't repeat yourself
Destructure values from `this` and reuse it
Fix all repeatable code and use destructuring